edx-platform/pearson-pols-theme/lms/templates/header/user_dropdown.html

```diff
@@ -2,38 +2,43 @@
 <%page expression_filter="h"/>
 <%namespace name='static' file='static_content.html'/>
 
-## This template should not use the target student's details when masquerading, see TNL-4895
-<%
-self.real_user = getattr(user, 'real_user', user)
-%>
-
 <%!
-from django.core.urlresolvers import reverse
+from django.urls import reverse
 from django.utils.translation import ugettext as _
 
+
 from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
 from openedx.core.djangoapps.user_api.accounts.image_helpers import get_profile_image_urls_for_user
+from openedx.core.djangoapps.user_api.accounts.utils import retrieve_last_sitewide_block_completed
+from openedx.features.enterprise_support.utils import get_enterprise_learner_generic_name
 %>
 
 <%
-    profile_image_url = get_profile_image_urls_for_user(self.real_user)['medium']
-    username = self.real_user.username
+## This template should not use the target student's details when masquerading, see TNL-4895
+self.real_user = getattr(user, 'real_user', user)
+profile_image_url = get_profile_image_urls_for_user(self.real_user)['medium']
+username = self.real_user.username
+resume_block = retrieve_last_sitewide_block_completed(self.real_user)
+displayname = get_enterprise_learner_generic_name(request) or username
 %>
 
-
 <div class="nav-item hidden-mobile">
     <a href="${reverse('dashboard')}" class="menu-title">
         <img class="user-image-frame" src="${profile_image_url}" alt="">
         <span class="sr-only">${_("Dashboard for:")}</span>
-        <span class="username">${username}</span>
+        <span class="username">${displayname}</span>
```

Comment:
```suggestion
        <span class="username">${display_name}</span>
```

Reply:
I will keep it in order to avoid differences with the original file